storage/namespace.go

```diff
@@ -537,8 +582,10 @@ func (n *dbNamespace) NeedsFlush(blockStart time.Time) bool {
 		if shard == nil {
 			continue
 		}
-		if shard.FlushState(blockStart).Status == fileOpInProgress {
-			return false
+		for _, blockStart := range blockStarts {
```

Comment:
This method is overloaded. We use it in two places:
(1) in the [flush manager](https://github.com/m3db/m3db/blob/c93ecb1dc6806088f616b5ff8cf76960e9acba90/storage/flush.go#L127-L132) to find the blocks the namespace needs to flush
(2) in the [cleanup manager](https://github.com/m3db/m3db/blob/c93ecb1dc6806088f616b5ff8cf76960e9acba90/storage/cleanup.go#L151-L167) to find the range of times the namespace has yet to flush and uses it to cleanup the commit log. 

Worse still, if we ever allow flushing and cleanup at the same time, we can loose data. Example scenario - a namespace is flushing a particular shard for block start `t0`, and the cleanup manager kicks in and tries to cleanup the commit log for a range including `t0`. It'll ask the namespace if it needs to flush that time, the namespace will see a running flush and return `false`. At which point the commit log for `t0` can be cleaned up despite a pending flush for that time. 

Removing the `fileOpInProgress` check eliminates this issue. Unless there's a strong reason to retain it, I'd like to remove it and rely on the flush manager to ensure only a single flush is happening at any time.